pkg/runtime/handler/docker_runtime.go

```diff
@@ -19,66 +19,87 @@ package handler
 import (
 	"context"
 	"fmt"
+	"net/http"
 	"os"
+	"path/filepath"
 	"strings"
 	"time"
 
 	"github.com/docker/docker/api/types/container"
-	"github.com/moby/moby/client"
+	dclient "github.com/docker/docker/client"
+
+	"github.com/koordinator-sh/koordinator/pkg/util/system"
+)
+
+var (
+	DockerEndpoint = filepath.Join(system.Conf.VarRunRootDir, "docker.sock")
 )
 
+var GetDockerClient = createDockerClient // for test
+
 type DockerRuntimeHandler struct {
-	dockerClient *client.Client
+	dockerClient *dclient.Client
 	endpoint     string
 }
 
 func NewDockerRuntimeHandler(endpoint string) (ContainerRuntimeHandler, error) {
-	dockerClient, err := createDockerClient(endpoint)
+	var (
+		client     *dclient.Client
+		httpClient *http.Client
+		err        error
+	)
+
+	client, err = GetDockerClient(httpClient, endpoint)
 	if err != nil {
 		return nil, err
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
 	defer cancel()
-	dockerClient.NegotiateAPIVersion(ctx)
+
+	client.NegotiateAPIVersion(ctx)
 
 	return &DockerRuntimeHandler{
-		dockerClient: dockerClient,
+		dockerClient: client,
 		endpoint:     endpoint,
 	}, err
 }
 
-func createDockerClient(endpoint string) (*client.Client, error) {
-	ep := strings.TrimPrefix(endpoint, "unix://")
-	if _, err := os.Stat(ep); err != nil {
+func createDockerClient(httpClient *http.Client, endPoint string) (*dclient.Client, error) {
+	ep := strings.TrimPrefix(endPoint, "unix://")
+
+	_, err := os.Stat(ep)
+	if err != nil {
```

Comment:
make it one line `if _, err := os.Stat(ep); err != nil`

Reply:
done